src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SyntaxEquivalentNode.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Interop
+{
+    public record struct SyntaxEquivalentNode<T>(T Node)
+        where T : SyntaxNode
+    {
+        public bool Equals(SyntaxEquivalentNode<T> other)
+        {
+            return SyntaxEquivalentComparer.Instance.Equals(Node, other.Node);
```

Comment:
I know the singleton is usually named Instance, but in this case I first thought this would compare the SyntaxNodes by reference/instance equality. I'm fine leaving it, but if you think it might be confusing too, it could be worth renaming.

Reply:
We can rename it, but I'd rather wait until after this PR is merged in.